File: src/types/teacher.ts
import type { Sabaq } from './lesson';
import type { PracticeAttempt } from './practice';
import type { TestSession } from './test';

/** Roster-list view of a student — enough for the teacher's overview screen. */
export interface Student {
  id: string;
  /** The underlying account id — distinct from `id` (the student *profile* id). Use this, not `id`, for anything that talks to a user (e.g. messaging). */
  userId: string;
  name: string;
  classId: string;
  currentStreak: number;
  todaysSabaq: Sabaq | null;
}

export interface TeacherFeedback {
  id: string;
  studentId: string;
  note: string;
  createdAt: string; // ISO datetime
}

/**
 * Full detail view for one student. Reuses PracticeAttempt/TestSession as-is
 * (same shapes the student's own Practice/Test pages use) rather than
 * inventing parallel teacher-side types — a teacher reviewing a student's
 * history should see the exact same record shape the student sees.
 */
export interface StudentDetail extends Student {
  recentPracticeAttempts: PracticeAttempt[];
  recentTestSessions: TestSession[];
  feedback: TeacherFeedback[];
}

File: backend/app/schemas/teacher.py
from datetime import datetime

from app.schemas.base import CamelModel
from app.schemas.lesson import SabaqOut
from app.schemas.practice import PracticeAttemptOut
from app.schemas.test import TestSessionOut


class StudentOut(CamelModel):
    id: str
    user_id: str
    name: str
    class_id: str | None
    current_streak: int
    todays_sabaq: SabaqOut | None


class TeacherClassOut(CamelModel):
    id: str
    name: str


class TeacherFeedbackOut(CamelModel):
    id: str
    student_id: str
    note: str
    created_at: datetime


class StudentDetailOut(StudentOut):
    recent_practice_attempts: list[PracticeAttemptOut]
    recent_test_sessions: list[TestSessionOut]
    feedback: list[TeacherFeedbackOut]


class AddFeedbackRequest(CamelModel):
    note: str
